### DISCUSSION TITLE
Hello, I think, target type of generic version of MvxTargetBinding<TTarget, TValue> is incorrect

### DISCUSSION THREAD

**Discussion by ivmazurenko** [MvvmCross/MvvmCross#4385] [General]
https://github.com/MvvmCross/MvvmCross/blob/1c002b3ba5148e10822ddaaade1a0a5b9fb0adfa/MvvmCross/Binding/Bindings/Target/MvxTargetBinding.cs#L70

Please, take a look for this line. Maybe i am wrong, but this property is used only once, when default value will be created in `MvxFullBinding`:

https://github.com/MvvmCross/MvvmCross/blob/1c002b3ba5148e10822ddaaade1a0a5b9fb0adfa/MvvmCross/Binding/Bindings/MvxFullBinding.cs#L170

Thus, I think, this line should be:
``
public Type TargetType => typeof(TValue);
``

Because it is type of value that will be applyed for target.

Can anybody explain me, why it is done this way?

**Comment by Cheesebaron** [comment]
TargetType is the type that the Target accepts. I.e.:

`UILabel.Text` which would be `string`

Why would you change that to use `TValue` which does not guarantee the same type?

**Reply by ivmazurenko** [reply]
I believe, I understands you correctly. Did you mean, that `UILabel` accepts `string` ? 

If yes, please, take a look following example:

https://github.com/MvvmCross/MvvmCross/blob/1c002b3ba5148e10822ddaaade1a0a5b9fb0adfa/MvvmCross/Platforms/Android/Binding/Target/MvxBaseViewVisibleBinding.cs#L20

This binding has `TargetType` `bool` (View accepts booleans in other words) because this class extends non-generic `MvxAndroidTargetBinding`. When `MvxFullBinding` will try to set default value for this target binding, it will use `boolean` value:
https://github.com/MvvmCross/MvvmCross/blob/1c002b3ba5148e10822ddaaade1a0a5b9fb0adfa/MvvmCross/Binding/Bindings/MvxFullBinding.cs#L161

But when we try to derive `MvxBaseViewVisibleBinding` from generic `MvxAndroidTargetBinding<View, bool> `, `TargetType` becomes `View`, because it is hardcoded in base class:

``
public Type TargetType => typeof(TTarget); 
``

And when MvxFullBinding will try to set default value here:

https://github.com/MvvmCross/MvvmCross/blob/1c002b3ba5148e10822ddaaade1a0a5b9fb0adfa/MvvmCross/Binding/Bindings/MvxFullBinding.cs#L170

It will try to put instance of `View` into the `Visibility` propery of this view, and this is not correct.

**Comment by ivmazurenko** [comment]
https://github.com/MvvmCross/MvvmCross/pull/4386
